File: app/src/main/java/org/autojs/autojs/model/project/ProjectTemplate.java
package org.autojs.autojs.model.project;

import android.annotation.SuppressLint;

import com.stardust.autojs.project.ProjectConfigKt;
import com.stardust.pio.PFiles;

import java.io.File;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class ProjectTemplate {


    private final ProjectConfigKt mProjectConfig;
    private final File mProjectDir;

    public ProjectTemplate(ProjectConfigKt projectConfig, File projectDir) {
        mProjectConfig = projectConfig;
        mProjectDir = projectDir;
    }

    @SuppressLint("CheckResult")
    public Observable<File> newProject() {
        return Observable.fromCallable(() -> {
            mProjectDir.mkdirs();
            PFiles.write(ProjectConfigKt.Companion.configFileOfDir(mProjectDir.getPath()), mProjectConfig.toJson());
            new File(mProjectDir, mProjectConfig.getMainScriptFile()).createNewFile();
            return mProjectDir;
        })
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
    }

}

File: app/src/main/java/org/autojs/autojs/model/project/ProjectTemplate.kt
package org.autojs.autojs.model.project

import com.stardust.autojs.project.ProjectConfig
import com.stardust.autojs.project.ProjectConfig.Companion.configFileOfDir
import com.stardust.pio.PFiles.write
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.io.File

class ProjectTemplate(private val mProjectConfig: ProjectConfig, private val mProjectDir: File) {

    suspend fun newProject(): File = withContext(Dispatchers.IO) {
        mProjectDir.mkdirs()
        write(configFileOfDir(mProjectDir.path), mProjectConfig.toJson())
        File(mProjectDir, mProjectConfig.mainScript!!).createNewFile()
        mProjectDir
    }
}